From a218b55610f01894691f71317a588f7ef88857ff Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 22 Jan 2024 10:45:03 +0100
Subject: [PATCH 075/120] rng: stm32: update STM32MP15 max RNG clock frequency

RNG max clock frequency can be updated to 48MHz according to latest
tests.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 7757a0e8c1be85521c1c662247f6043e3ceb8799 rng: stm32: update STM32MP15 max RNG clock frequency]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I8b460e83b72ed7c9e9a40535ecdf23a978f98772
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/353186
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/359775
Tested-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/rng/stm32_rng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/rng/stm32_rng.c b/drivers/rng/stm32_rng.c
index aff0b4915e8..27ec0bd9d83 100644
--- a/drivers/rng/stm32_rng.c
+++ b/drivers/rng/stm32_rng.c
@@ -420,7 +420,7 @@ static const struct stm32_rng_data stm32mp13_rng_data = {
 
 static const struct stm32_rng_data stm32_rng_data = {
 	.has_cond_reset = false,
-	.max_clock_rate = 3000000,
+	.max_clock_rate = 48000000,
 	.nb_clock = 1,
 };
 
-- 
2.47.2

